torch/_inductor/lowering.py

```diff
@@ -690,6 +703,50 @@ def _to_dtype(x):
     return make_pointwise(_to_dtype, override_return_dtype=dtype)(x)
 
 
+@register_lowering(torch._higher_order_ops._foreach_map)
+def _foreach_map(subgraph, *args, **kwargs):
+    realize_outputs = (
+        len(V.graph.current_node.users) == 0 or cur_node_has_non_foreach_users()
+    )
+
+    from .subgraph_lowering import PointwiseSubgraphLowering
+
+    inputs = args[0]  # nested tuple
+
+    gm = subgraph.graph_module
+    pw_subgraph = PointwiseSubgraphLowering(gm, root_graph_lowering=V.graph)
```

Comment:
we don't do this in the original foreach lowerings as far as i can tell. would you say more about why we're doing it here ?

Reply:
Yeah I can add a comment, this is the main difference from the foreach lowering. The way this works is that an arbitrary N-arg func is provided by the user, looped over by the polyfill and then traced into a subgraph by dynamo. This code allows us to inline the subgraph into the main graph lowering using the PontwiseSubgraphLowering. The graph outputs represent the vertically fused sequence of ops, and then `register_operation_list` below registers the buffers as horizontally fuseable in the scheduler.

Reply:
Yea, please add comment.